app/controllers/application_controller.rb

```diff
@@ -16,8 +16,15 @@ def authorize_admin!
      end
   end
 
+  def authorize_donor!
+     unless current_donor && !current_donor.admin
```

Comment:
Just as an FYI - when using booleans like this, it's customary to put a `?` on it. So:
`unless current_donor && !current_donor.admin?`